R/qenv-eval_code.R

```diff
@@ -31,7 +31,7 @@ setMethod("eval_code", signature = c("qenv", "character"), function(object, code
   id <- sample.int(.Machine$integer.max, size = 1)
 
   object@id <- c(object@id, id)
-  object@env <- rlang::env_clone(object@env, parent = parent.env(.GlobalEnv))
+  object@.xData <- rlang::env_clone(object@.xData, parent = parent.env(.GlobalEnv))
```

Comment:
{rlang} functions don't play nicely with this approach, so we need to access the slot directly

``` r
pkgload::load_all("teal.code")
#> ℹ Loading teal.code

q1 <- qenv()
rlang::is_environment(q1)
#> [1] FALSE
base::is.environment(q1)
#> [1] TRUE

rlang::env_clone(q1)
#> Error in `rlang::env_clone()`:
#> ! `env` must be an environment, not a <qenv> object.

rlang::is_environment
#> function(x) {
#>   typeof(x) == "environment"
#> }
#> <bytecode: 0x65127a0d6b08>
#> <environment: namespace:rlang>
typeof(q1)
#> [1] "S4"
```

<sup>Created on 2024-10-29 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
Not a blocker for this approach 👍